test/02/t0262a.sh

```diff
@@ -0,0 +1,34 @@
+#!/bin/sh
+#
+# srecord - Manipulate EPROM load files
+# Copyright (C) 2014 Markus Heidelberg
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or (at
+# your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+
+TEST_SUBJECT="-Round_Up"
+. test_prelude.sh
+
+cat > test.ok << 'fubar'
```

Comment:
I would have guessed, the test will fail because `test.in` could not be found, because here the wrong file is created. But it seems the new error message comes before trying to read the file.

How about rewrite the whole test, and check the error message:

~~~sh

srec_cat -generate 0 8 -constant 0 -offset - 1 -Round_Up 0 \
  2> test.out

if test $? -ne 1; then fail; fi

cat > test.ok << 'fubar'
srec_cat: -Round_Up value must not be 0
fubar
if test $? -ne 0; then no_result; fi

diff -u test.ok test.out
if test $? -ne 0; then fail; fi
~~~

- The input is irrelevant, so just use some constants.
- The output would be irrelevant, so not write it to a file (that gets not generated, because of the error message).
- Write the error message to a file, and test if it's the expected one occured

Reply:
All three tests now generate random data and validate printed error message.